src/dfx/src/actors/mod.rs

```diff
@@ -54,7 +58,10 @@ pub fn start_emulator_actor(
     env: &dyn Environment,
     shutdown_controller: Addr<ShutdownController>,
 ) -> DfxResult<Addr<Emulator>> {
-    let ic_ref_path = env.get_cache().get_binary_command_path("ic-ref")?;
+    let ic_ref_path = env
+        .get_cache()
+        .get_binary_command_path("ic-ref")
+        .context("Failed to fetch 'ic-ref' binary.")?;
```

Comment:
Would you please take a look at `impl Cache for DiskBasedCache {
` in src/config/cache.rs?  It might provide a single place to add this particular type of context.  WDYT?